client/src/Atoms/UserProfileData/UserProfileData.js

```diff
@@ -38,19 +37,26 @@ const UserProfileData = ({ withEdit, initialState }) => {
   const { address, participantName, participantSurname } = userObject;
   const { street, numberOfHouse, city, postalCode } = address;
 
-  // mock for fetching data from database and checking if response has data
-
   useEffect(() => {
     if (!isAuthenticated) {
       navigate('/login');
-    }
-    const userObject = participants.find(
-      (participant) => participant.participantId === userData,
-    );
-    if (!userObject) {
-      navigate('/login');
     } else {
-      setUserObject(userObject);
+      const requestOptions = {
+        method: 'GET',
+        redirect: 'follow',
+        credentials: 'include',
+      };
+
+      fetch(`http://localhost:27020/api/users/${userData}`, requestOptions)
+        .then((response) => response.json())
+        .then((result) => {
+          if (!result) {
+            navigate('/login');
+          } else {
+            setUserObject(result);
+          }
+        })
+        .catch((error) => console.log('error', error));
```

Comment:
Temat do przemyslenia - error handling na froncie

Reply:
Dopiszę jako issue, bo to kolejny duży temat :)